crates/client/rpc/src/errors.rs

```diff
@@ -42,6 +45,15 @@ pub enum StarknetRpcApiError {
     UnimplementedMethod = 501,
 }
 
+impl From<DispatchError> for StarknetRpcApiError {
+    fn from(err: DispatchError) -> Self {
+        match err {
+            _ if err == PalletError::<Runtime>::ContractNotFound.into() => StarknetRpcApiError::ContractNotFound,
```

Comment:
Don't use `into` on the right arm of you comparison. 
Just use the thing err is actually equal to

Reply:
In this context, in order to compare the given `err: DispatchError` with the [pallet error](https://github.com/keep-starknet-strange/madara/blob/8bba0040b29f09e817d3205a6c1410e4b77d49d1/crates/pallets/starknet/src/lib.rs#L441), I think that explicit conversion with `into` is required. Is there another way to do this without using `into` that I'm missing ?

Reply:
There is a way, you can manualy create the struct:
```rust
DispatchError::Module(ModuleError{...})
```
But it will be unmaintainable and unreadable. So keep it the way it is.